app/fileManager.js

```diff
@@ -51,6 +52,19 @@ function exists(id) {
     xhr.send();
   });
 }
+*/
+
+async function getDLCounts(file) {
+  const url = `/api/metadata/${file.id}`;
+  const receiver = new FileReceiver(url, file);
+  try {
+    await receiver.getMetadata(file.nonce);
+    return receiver.file;
+  } catch (e) {
+    console.log(e);
+    return null;
```

Comment:
This should actually be `return true` for the condition to work 
```
const receivedFile = await getDLCounts(file);
 if (receivedFile) {
```

Reply:
Is this a question? ;)

If this case returned `true` then both possible return values would be "truthy". i.e. `if(result)` would *always* be true.

Take a step back and look at the function, `getDLCounts(file)`. If you didn't know what it did, what would you expect / want it to return? Maybe a number? Maybe a Map or Object with multiple numbers (named dtotal and dlimit)? Now what about if the `file` wasn't valid anymore? Should it return a special value or throw an error? Think about how you'd like to *consume* this function, then try to make it behave that way.

Reply:
Yeah I've got this. Thanx. Please have a look now.